### ISSUE TITLE
Inconsistent handling of rejected queries in Cosmos DB storage

### ISSUE THREAD

**Issue by johnstairs** [microsoft/fhir-server#451]
In upsertwithhistory.js and harddelete.js, we employ this pattern:
``` js
let isAccepted = collection.replaceDocument(selfLink, doc, { disableAutomaticIdGeneration: true, etag: document._etag }, callback);

if (!isAccepted) {
    throw createRequestNotQueuedError();
}
```

But we throw different error codes. We should see what the error code should be and whether 429 is an option to allow the SDK to perform a retry.

**Comment by CaitlinV39** [comment]
@johnstairs Is this issue fully resolved by #449 or is there still work to do?

**Comment by johnstairs** [comment]
@CaitlinV39, this is fully resolved. Closing.
